parser/src/format/projection.rs

```diff
@@ -37,17 +44,90 @@ pub enum BuildError {
     SchemaIndex(#[from] SchemaIndexError),
 }
 
-/// Map input characters (e.x. String::chars()) into its collated form,
-/// which ignores casing and is unicode-normalized.
-/// This follows the conformance guidelines in:
-/// http://www.unicode.org/versions/Unicode13.0.0/ch03.pdf
-/// in Section 3.13 - "Default Caseless Matching" (all the way at the bottom).
-// This must match the collation used by Flow during the catalog build process.
-pub fn collate<I>(i: I) -> impl Iterator<Item = char>
-where
-    I: Iterator<Item = char>,
-{
-    i.nfd().default_case_fold().nfkc()
+/// Projections is a helper for constructing potentially nested JSON documents from flat tabular
+/// input.
+#[derive(Debug)]
+pub struct Projections {
+    /// Type inferrence information from the root JSON schema.
+    shape: Shape,
+    /// Projections stored with field names exactly as they are in the config/schema.
+    case_sensitive: BTreeMap<String, Rc<Pointer>>,
+    /// Projections stored with field names that have been normalized and converted to lowercase,
+    /// to allow for case-insensitive matching against field names. Technically, these _could_ all
+    /// be included in one big map, but
```

Comment:
I think the end of this comment got lost...

Reply:
whoops, fixed.